spring-pulsar/src/main/java/org/springframework/pulsar/core/TypedMessageBuilderCustomizer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.core;
+
+import org.apache.pulsar.client.api.TypedMessageBuilder;
+
+/**
+ * The interface to customize a {@link TypedMessageBuilder}.
+ *
+ * @author Alexander Preuß
+ */
+@FunctionalInterface
+public interface TypedMessageBuilderCustomizer {
```

Comment:
Wdyt about adding the type `<T>` here and parameterize the parameter below w/ it?